### DISCUSSION TITLE
How do I find the challenge contracts and when do I know I am ready to do them?

### DISCUSSION THREAD

**Discussion by FelipeCevallos** [Cyfrin/foundry-full-course-cu#2274] [Q&A] [answered]
Hey guys! I am currently completing the Cyfrin Updraft course "Solidity Smart Contract Development" and after finishing Section 2: Storage Factory, the video says that I should try to complete the challenge contracts. The written lesson also states: "Attempt to answer all the theoretical questions from lesson 1 through 7, and then go back again to complete all the coding tasks.". 

I have no idea where the challenge contracts are or when I am supposed to attempt completing them. I would love to put in practice the knowledge learned during these lessons.

I have found this post: https://github.com/Cyfrin/foundry-full-course-cu/discussions/135
But I have no clue as to whether or not they are related to the sections of the SC Development course. 

Any help will be greatly appreciated!!

**Comment by EngrPips** [comment, marked as answer]
Hello @FelipeCevallos, Read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2018 from `Foundry-Full-Course` I explained in detail how to get the challenge contract for each section in any Cyfrin course.

**Reply by FelipeCevallos** [reply]
Absolute wizard! Thank you so much!!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
